### ISSUE TITLE
Non-determinism in SC TF across runs

### ISSUE THREAD

**Issue by fareeha** [apple/turicreate#2788]
ESC-10 dataset (passing audio data and pre-computed deep feature data)

The per iteration accuracy varies across multiple runs in TF. On MXNet the convergence is identical.

**TF using deep features as input**

**Run 1**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.846             | 35.977            |
+-------------------+-------------------+-------------------+
| 2                 | 0.922             | 36.266            |
+-------------------+-------------------+-------------------+
| 3                 | 0.930             | 36.564            |
+-------------------+-------------------+-------------------+
| 4                 | 0.938             | 36.862            |
+-------------------+-------------------+-------------------+
| 5                 | 0.944             | 37.161            |
+-------------------+-------------------+-------------------+
| 6                 | 0.951             | 37.460            |
+-------------------+-------------------+-------------------+
| 7                 | 0.945             | 37.733            |
+-------------------+-------------------+-------------------+
| 8                 | 0.953             | 38.006            |
+-------------------+-------------------+-------------------+
| 9                 | 0.953             | 38.277            |
+-------------------+-------------------+-------------------+
| 10                | 0.946             | 38.548            |
+-------------------+-------------------+-------------------+
```
**Run 2**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.792             | 32.647            |
+-------------------+-------------------+-------------------+
| 2                 | 0.891             | 32.919            |
+-------------------+-------------------+-------------------+
| 3                 | 0.912             | 33.184            |
+-------------------+-------------------+-------------------+
| 4                 | 0.948             | 33.456            |
+-------------------+-------------------+-------------------+
| 5                 | 0.947             | 33.729            |
+-------------------+-------------------+-------------------+
| 6                 | 0.906             | 34.002            |
+-------------------+-------------------+-------------------+
| 7                 | 0.944             | 34.274            |
+-------------------+-------------------+-------------------+
| 8                 | 0.953             | 34.554            |
+-------------------+-------------------+-------------------+
| 9                 | 0.954             | 34.828            |
+-------------------+-------------------+-------------------+
| 10                | 0.946             | 35.115            |
+-------------------+-------------------+-------------------+
```
**TF using audio data as input**
**Run 1**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.856             | 222.073           |
+-------------------+-------------------+-------------------+
| 2                 | 0.901             | 222.357           |
+-------------------+-------------------+-------------------+
| 3                 | 0.937             | 222.641           |
+-------------------+-------------------+-------------------+
| 4                 | 0.948             | 222.922           |
+-------------------+-------------------+-------------------+
| 5                 | 0.947             | 223.201           |
+-------------------+-------------------+-------------------+
| 6                 | 0.941             | 223.479           |
+-------------------+-------------------+-------------------+
| 7                 | 0.946             | 223.760           |
+-------------------+-------------------+-------------------+
| 8                 | 0.946             | 224.041           |
+-------------------+-------------------+-------------------+
| 9                 | 0.946             | 224.325           |
+-------------------+-------------------+-------------------+
| 10                | 0.946             | 224.601           |
+-------------------+-------------------+-------------------+
```
**Run 2**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.831             | 223.929           |
+-------------------+-------------------+-------------------+
| 2                 | 0.853             | 224.210           |
+-------------------+-------------------+-------------------+
| 3                 | 0.919             | 224.487           |
+-------------------+-------------------+-------------------+
| 4                 | 0.930             | 224.766           |
+-------------------+-------------------+-------------------+
| 5                 | 0.948             | 225.047           |
+-------------------+-------------------+-------------------+
| 6                 | 0.932             | 225.324           |
+-------------------+-------------------+-------------------+
| 7                 | 0.939             | 225.605           |
+-------------------+-------------------+-------------------+
| 8                 | 0.947             | 225.886           |
+-------------------+-------------------+-------------------+
| 9                 | 0.947             | 226.164           |
+-------------------+-------------------+-------------------+
| 10                | 0.947             | 226.444           |
+-------------------+-------------------+-------------------+
```

**MXNET using deep features as input**

**Run 1** 
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.371             | 37.684            |
+-------------------+-------------------+-------------------+
| 2                 | 0.591             | 37.969            |
+-------------------+-------------------+-------------------+
| 3                 | 0.677             | 38.244            |
+-------------------+-------------------+-------------------+
| 4                 | 0.781             | 38.520            |
+-------------------+-------------------+-------------------+
| 5                 | 0.820             | 38.791            |
+-------------------+-------------------+-------------------+
| 6                 | 0.846             | 39.061            |
+-------------------+-------------------+-------------------+
| 7                 | 0.864             | 39.333            |
+-------------------+-------------------+-------------------+
| 8                 | 0.886             | 39.599            |
+-------------------+-------------------+-------------------+
| 9                 | 0.900             | 39.871            |
+-------------------+-------------------+-------------------+
| 10                | 0.911             | 40.150            |
+-------------------+-------------------+-------------------+
```
**Run 2**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.371             | 36.252            |
+-------------------+-------------------+-------------------+
| 2                 | 0.591             | 36.524            |
+-------------------+-------------------+-------------------+
| 3                 | 0.677             | 36.797            |
+-------------------+-------------------+-------------------+
| 4                 | 0.781             | 37.077            |
+-------------------+-------------------+-------------------+
| 5                 | 0.820             | 37.355            |
+-------------------+-------------------+-------------------+
| 6                 | 0.846             | 37.635            |
+-------------------+-------------------+-------------------+
| 7                 | 0.864             | 37.912            |
+-------------------+-------------------+-------------------+
| 8                 | 0.886             | 38.181            |
+-------------------+-------------------+-------------------+
| 9                 | 0.900             | 38.445            |
+-------------------+-------------------+-------------------+
| 10                | 0.911             | 38.715            |
+-------------------+-------------------+-------------------+
```
**MXNET using audio data as input** 
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.371             | 259.942           |
+-------------------+-------------------+-------------------+
| 2                 | 0.591             | 260.199           |
+-------------------+-------------------+-------------------+
| 3                 | 0.677             | 260.480           |
+-------------------+-------------------+-------------------+
| 4                 | 0.781             | 260.770           |
+-------------------+-------------------+-------------------+
| 5                 | 0.820             | 261.036           |
+-------------------+-------------------+-------------------+
| 6                 | 0.846             | 261.291           |
+-------------------+-------------------+-------------------+
| 7                 | 0.864             | 261.538           |
+-------------------+-------------------+-------------------+
| 8                 | 0.886             | 261.795           |
+-------------------+-------------------+-------------------+
| 9                 | 0.900             | 262.046           |
+-------------------+-------------------+-------------------+
| 10                | 0.911             | 262.296           |
+-------------------+-------------------+-------------------+
```

**Comment by fareeha** [comment]
^ MXNET and TF from 6.0
(MXNET in 5.8 shows non determinism as well)

**Comment by fareeha** [comment]
Closed in favor of #2806
